File: tools/bench/answers/go_sonnet/s2/15_wordlens.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	sentence := "pack my box with five dozen liquor jugs"
	words := strings.Fields(sentence)
	longest := ""
	totalLen := 0
	for _, w := range words {
		totalLen += len(w)
		if len(w) > len(longest) {
			longest = w
		}
	}
	fmt.Println(longest)
	fmt.Println(totalLen / len(words))
}

File: tools/bench/answers/rust_haiku/s2/15_wordlens.rs
fn main() {
    let sentence = "pack my box with five dozen liquor jugs";
    let words: Vec<&str> = sentence.split_whitespace().collect();

    let longest = words.iter().max_by_key(|w| w.len()).unwrap();

    let total_len: usize = words.iter().map(|w| w.len()).sum();
    let avg = total_len as i32 / words.len() as i32;

    println!("{}", longest);
    println!("{}", avg);
}
